docs/gatsby-node.js

```diff
@@ -0,0 +1,46 @@
+const path = require("path");
+
+// creates a "collection" field to it easier to filter nodes created by
+// gatsby-source-filesystem instead of awkwardly filtering by file path
+// https://github.com/gatsbyjs/gatsby/issues/1634#issuecomment-388899348
+exports.onCreateNode = ({ node, getNode, actions }) => {
+  if (node.internal.type !== "Mdx") return;
+  const { createNodeField } = actions;
+  const parent = getNode(node.parent);
+  createNodeField({
+    node,
+    name: "collection",
+    value: parent.sourceInstanceName,
+  });
+};
+
+exports.createPages = async ({ graphql, actions, reporter }) => {
+  const { createPage } = actions;
+  const result = await graphql(`
+    query AllGuides {
+      allMdx(filter: { fields: { collection: { eq: "guides" } } }) {
+        nodes {
+          frontmatter {
+            title
+          }
+          body
+          slug
+        }
+      }
+    }
+  `);
+  if (result.errors) {
+    reporter.panicOnBuild(`Error when querying guides.`);
+    return;
+  }
+  result.data.allMdx.nodes.forEach(guide => {
+    createPage({
+      path: `/guides/${path.basename(guide.slug)}`,
```

Comment:
This renders, e.g., `{base}/guides/using-component-types`. Shouldn't it be `{base}/developer/using-component-types`? Two things here:
1. I don't see the purpose of "guides" in the url.
2. I think the collection should be a part of the url.

As an aside, I prefer it with the trailing slash. Right now, I think both versions are being generated (because with trailing slash is done automatically), which can appear like duplicated content for SEO purposes (maybe?).

Reply:
> This renders, e.g., `{base}/guides/using-component-types`. Shouldn't it be `{base}/developer/using-component-types`? 

The decision was based solely on the assumption that we want to retain the same URLs, see https://orbit.kiwi/guides/using-component-types/

> As an aside, I prefer it with the trailing slash. Right now, I think both versions are being generated (because with trailing slash is done automatically), which can appear like duplicated content for SEO purposes (maybe?).

Thanks for the note, I'll definitely look into that.

Reply:
I think we're going to restructure and I'd prefer the URLs to match the new structure. I'd add [`gatsby-redirect-from`](https://github.com/kremalicious/gatsby-redirect-from) to keep the old URLs functional, but I wouldn't necessarily stick to them for the future.

Reply:
Got it! I wasn't aware that this was the plan.